lib_eio_gcd/dqueue.ml

```diff
@@ -0,0 +1,92 @@
+(* A read/write request and completion queue inspired by io_uring *)
+type 'a rw = {
+  req : [ `R | `W ];
+  fd : Unix.file_descr;
+  file_offset : int;
+  len : int;
+  id : int;
+  buff : Dispatch.Data.t ref;
+  data : 'a;
+}
+
+type 'a request = Close of Unix.file_descr * 'a | Rw of 'a rw
+
+type eof = Eof | Eof_with_data | Not_eof
```

Comment:
This type could probably use a little more explanation. I guess there's no `Not_eof_with_data` since that circumstance can never happen, but could it with a TCP half close when the other side has indicated it'll never send more data after the current chunk is complete?